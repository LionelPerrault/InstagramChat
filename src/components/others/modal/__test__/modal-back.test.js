import React from 'react'
import ModalBack from '../modal-back'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('ModalBack Component', () => {
  const mockFn = jest.fn()

  it('should match snapshot with primary button', () => {
    const tree = create(<ModalBack back={mockFn} />).toJSON()
    expect(tree).toMatchSnapshot()

    const wrapper = shallow(<ModalBack back={mockFn} />)
    wrapper.find('PrimaryButton').simulate('click', { preventDefault() {} })
  })

  it('should match snapshot with secondary button', () => {
    const tree = create(
      <ModalBack back={mockFn} btnType="secondary" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
