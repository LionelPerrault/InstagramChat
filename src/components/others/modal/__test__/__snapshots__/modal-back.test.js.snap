// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBack Component should match snapshot with primary button 1`] = `
<a
  className="pri_btn"
  href="#"
  onClick={[Function]}
>
  Back
</a>
`;

exports[`ModalBack Component should match snapshot with secondary button 1`] = `
<a
  className="sec_btn"
  href="#"
  onClick={[Function]}
>
  Back
</a>
`;
